data-prepper-plugins/aws-sqs-common/src/main/java/org/opensearch/dataprepper/plugins/aws/sqs/common/SqsService.java

```diff
@@ -0,0 +1,203 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.opensearch.dataprepper.plugins.aws.sqs.common;
+
+import com.linecorp.armeria.client.retry.Backoff;
+import org.opensearch.dataprepper.model.acknowledgements.AcknowledgementSet;
+import org.opensearch.dataprepper.model.acknowledgements.AcknowledgementSetManager;
+import org.opensearch.dataprepper.plugins.aws.sqs.common.exception.SqsRetriesExhaustedException;
+import org.opensearch.dataprepper.plugins.aws.sqs.common.metrics.SqsMetrics;
+import org.opensearch.dataprepper.plugins.aws.sqs.common.model.SqsOptions;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import software.amazon.awssdk.services.sqs.SqsClient;
+import software.amazon.awssdk.services.sqs.model.DeleteMessageBatchRequest;
+import software.amazon.awssdk.services.sqs.model.DeleteMessageBatchRequestEntry;
+import software.amazon.awssdk.services.sqs.model.DeleteMessageBatchResponse;
+import software.amazon.awssdk.services.sqs.model.DeleteMessageBatchResultEntry;
+import software.amazon.awssdk.services.sqs.model.Message;
+import software.amazon.awssdk.services.sqs.model.ReceiveMessageRequest;
+import software.amazon.awssdk.services.sqs.model.ReceiveMessageResponse;
+import software.amazon.awssdk.services.sqs.model.SqsException;
+import software.amazon.awssdk.services.sts.model.StsException;
+
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ * contains the sqs related common functionality for reading
+ * the messages and processing
+ */
+public class SqsService {
+    private static final Logger LOG = LoggerFactory.getLogger(SqsService.class);
+
+    private final SqsMetrics sqsMetrics;
+
+    private final SqsClient sqsClient;
+
+    private final Backoff backoff;
+
+    private int failedAttemptCount;
+
+    public SqsService(final SqsMetrics sqsMetrics,
+                      final SqsClient sqsClient,
+                      final Backoff backoff) {
+        this.sqsMetrics = sqsMetrics;
+        this.sqsClient = sqsClient;
+        this.backoff =backoff;
+
+    }
+
+    /**
+     * Create a sqs message request object with the help of queue url and max messages count.
+     *
+     * @param sqsOptions - required sqs option object
+     * @return ReceiveMessageRequest - return the Aws Message Request object.
+     */
+    public ReceiveMessageRequest createReceiveMessageRequest(final SqsOptions sqsOptions) {
```

Comment:
Added a comment in issue. https://github.com/opensearch-project/data-prepper/issues/2679#issuecomment-1589620368